bittensor/extrinsics/delegation.py

```diff
@@ -126,19 +125,19 @@ def delegate_extrinsic(
     """
     # Decrypt keys,
     wallet.coldkey
-    if not subtensor.is_hotkey_delegate(delegate_ss58):
+
+    if not await subtensor.is_hotkey_delegate(delegate_ss58):
         raise NotDelegateError("Hotkey: {} is not a delegate.".format(delegate_ss58))
 
     # Get state.
-    my_prev_coldkey_balance = subtensor.get_balance(wallet.coldkey.ss58_address)
-    delegate_take = subtensor.get_delegate_take(delegate_ss58)
-    delegate_owner = subtensor.get_hotkey_owner(delegate_ss58)
-    my_prev_delegated_stake = subtensor.get_stake_for_coldkey_and_hotkey(
+    my_prev_coldkey_balance = await subtensor.get_balance(wallet.coldkey.ss58_address)
+    delegate_owner = await subtensor.get_hotkey_owner(delegate_ss58)
+    my_prev_delegated_stake = await subtensor.get_stake_for_coldkey_and_hotkey(
```

Comment:
do these need to be gathered before moving on?

Reply:
I thought about migrating first, and then apply asincoo.gather during the optimization + testing. Do you think it's worth it right away?

Reply:
Thats fine! Thanks for clearing up your thought process.